instrumentation/src/main/java/io/opentelemetry/android/emitter/EventEmitterWithData.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.emitter;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.api.events.EventEmitter;
+import java.util.Map;
+import java.util.Objects;
+
+class EventEmitterWithData {
+
+    private static final String EVENT_DATA = "event.data";
+
+    private final EventEmitter delegate;
+
+    EventEmitterWithData(EventEmitter delegate) {
+        this.delegate = delegate;
+    }
+
+    public void emit(String eventName, Attributes attributes, Map<String, Object> data) {
+        Attributes dataAsAttribute = convertRichDataObjectIntoAttributes(data);
+        Attributes eventAttrs =
+                attributes.toBuilder().put(EVENT_DATA, dataAsAttribute.toString()).build();
+        delegate.emit(eventName, eventAttrs);
+    }
+
+    private Attributes convertRichDataObjectIntoAttributes(Map<String, Object> data) {
+        AttributesBuilder builder = Attributes.builder();
+        for (String key : data.keySet()) {
+            Object val = data.get(key);
+            if (val != null) {
+                if (val instanceof String) {
+                    builder.put(key, (String) Objects.requireNonNull(data.get(key)));
```

Comment:
- As on line 35 it already checks that val != null, do we need Objects.requireNonNull here? 
- Objects.requireNonNull throws null pointer exception which might not be good.  I think just ignoring the attribute (map entry) if val is null is fine which is done on line 35. You can just use val here directly.

Reply:
Not sure why I didn't use val directly here, my bad. Corrected